test/unit/question_base_test.rb

```diff
@@ -5,6 +5,15 @@ class QuestionBaseTest < ActiveSupport::TestCase
     @question = SmartAnswer::Question::Base.new(nil, :example_question)
   end
 
+  context "#on_response" do
+    should "not allow multiple definitions of on_response" do
+      assert_raises do
```

Comment:
I think this should assert on the message as well so we have confidence that failure is occurring where expected.